server/service/src/login.rs

```diff
@@ -627,6 +639,33 @@ mod test {
 
             assert!(result.is_ok());
         }
+        // check login error handling when empty password hash and can't connect to mSupply
+        {
+            let mock_server = MockServer::start();
+            mock_server.mock(|when, then| {
+                when.method(POST).path("/api/v4/login".to_string());
+                then.status(500);
+            });
+
+            let central_server_url = mock_server.base_url();
+
+            let result = LoginService::login(
+                &service_provider,
+                &auth_data,
+                LoginInput {
+                    username: mock_user_empty_hashed_password().username,
+                    password: "password".to_string(),
+                    central_server_url,
+                },
+                0,
+            )
+            .await;
+
+            assert_matches!(
+                result,
+                Err(LoginError::MSupplyCentralNotReached)
+            );
+        }
```

Comment:
I wanted also here to add a check for internal error when the password was missing and mSupply central was reached, but I couldn't get the set up working

Reply:
Had a go here! https://github.com/msupply-foundation/open-msupply/compare/4425-add-better-error-messaging-for-unsynced-passwords...4425-add-better-error-messaging-for-unsynced-passwords(test-internal-error)

Reply:
That is epic! Thanks Laché!

I fear it won't test the exact use case of receiving an OK user_info from mSupply central rather than an internal error. But I'm guessing that would add complexity for simulating the response from mSupply.

Do you reckon worthwhile trying to add that use case?

Reply:
That _shouldn't_ really be possible...

You can see here, if fetch from central succeeds, we call update_user, which hashes the inputted password. If we somehow don't get a hashed password, something has gone rather awry! So I think fine to rely on internal error 😁 

https://github.com/msupply-foundation/open-msupply/blob/a2744c6d316dd1f2df938ba5df66a2bed93e69bb/server/service/src/login.rs#L118-L123